File: packages/dotnet/src/Resources/AccountResource.cs
using ImzalaApiClient.Api;
using ImzalaApiClient.Model;

namespace ImzalaSdk;

/// <summary>Backs <c>imzala.MeAsync()</c> — <c>ImzalaApiClient.Api.AccountApi</c> under the hood.</summary>
internal sealed class AccountResource
{
    private readonly IAccountApi _api;
    private readonly RetryConfig _retry;

    internal AccountResource(IAccountApi api, RetryConfig retry)
    {
        _api = api;
        _retry = retry;
    }

    /// <summary>Returns the calling API key's owner info (id, email, name, workspace, remaining credits). Requires the <c>timestamps</c> scope. GET — safe to auto-retry.</summary>
    public Task<ApiV1MeGet200ResponseData> MeAsync(CancellationToken cancellationToken = default) =>
        Http.UnwrapRetryableGet(() => _api.ApiV1MeGetAsync(cancellationToken), r => r.Success, r => r.Data, _retry);
}

File: packages/java/src/main/java/org/imzala/AccountResource.java
package org.imzala;

import org.imzala.client.generated.api.AccountApi;
import org.imzala.client.generated.model.ApiV1MeGet200ResponseData;

/** Backs {@code imzala.me()} — the vendored generated {@code AccountApi} under the hood. */
final class AccountResource {

  private final AccountApi api;

  AccountResource(AccountApi api) {
    this.api = api;
  }

  /** Returns the calling API key's owner info (id, email, name, workspace, remaining credits). Requires the {@code timestamps} scope. */
  ApiV1MeGet200ResponseData me() {
    return Http.unwrap(
        api::apiV1MeGet,
        r -> Boolean.TRUE.equals(r.getSuccess()),
        r -> r.getData());
  }
}
